webui/react/src/components/RunActionDropdown.tsx

```diff
@@ -0,0 +1,190 @@
+            handlePath(e as MouseEvent, { path: link, popout: 'tab' });
+            await onLink?.();
+            break;
+          case Action.NewWindow:
+            handlePath(e as MouseEvent, { path: link, popout: 'window' });
+            await onLink?.();
+            break;
+          case Action.Archive:
+            await archiveRuns({ projectId, runIds: [id] });
+            await onComplete?.(action, id);
+            break;
+          case Action.Kill:
+            confirm({
+              content: `Are you sure you want to kill run ${id}?`,
+              danger: true,
+              okText: 'Kill',
+              onConfirm: async () => {
+                await killRuns({ projectId, runIds: [id] });
+                await onComplete?.(action, id);
+              },
+              onError: handleError,
+              title: 'Confirm Run Kill',
+            });
+            break;
+          case Action.Unarchive:
+            await unarchiveRuns({ projectId, runIds: [id] });
+            await onComplete?.(action, id);
+            break;
+          case Action.Delete:
+            confirm({
+              content: `Are you sure you want to delete run ${id}?`,
+              danger: true,
+              okText: 'Delete',
+              onConfirm: async () => {
+                await deleteRuns({ projectId, runIds: [id] });
+                await onComplete?.(action, id);
+              },
+              onError: handleError,
+              title: 'Confirm Run Deletion',
+            });
+            break;
+          case Action.Move:
+            flatRunMoveModalOpen();
+            break;
+          case Action.Copy:
+            /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+            await copyToClipboard((cell as any).displayData || cell?.copyData);
+            openToast({
+              severity: 'Confirm',
+              title: 'Value has been copied to clipboard.',
+            });
+            break;
+        }
+      } catch (e) {
+        handleError(e, {
+          level: ErrorLevel.Error,
+          publicMessage: `Unable to ${action} experiment ${id}.`,
+          publicSubject: `${capitalize(action)} failed.`,
+          silent: false,
+          type: ErrorType.Server,
+        });
+      } finally {
+        onVisibleChange?.(false);
+      }
+    },
+    [
+      link,
+      onLink,
+      id,
+      onComplete,
+      confirm,
+      cell,
+      openToast,
+      onVisibleChange,
+      projectId,
+      flatRunMoveModalOpen,
+    ],
+  );
+
+  const shared = (
+    <>
+      <FlatRunMoveComponentModal
+        filterFormSetWithoutId={filterFormSetWithoutId}
+        flatRuns={[run]}
+        sourceProjectId={projectId}
+        sourceWorkspaceId={run.workspaceId}
+        onActionComplete={() => onComplete?.(FlatRunAction.Move, id)}
+      />
+    </>
+  );
+
+  return (
+    <>
+      <Dropdown isContextMenu menu={dropdownMenu} open={makeOpen} onClick={handleDropdown}>
+        <div />
```

Comment:
is this neeeded?

Reply:
we do need it to prevent error in `Dropdown` component -- could probably refactor the Dropdown component to make it not required but I think that would be a separate issue.

Reply:
interesting